control-plane/pkg/security/secrets_provider_legacy_channel_secret.go

```diff
@@ -29,7 +30,7 @@ func ResolveAuthContextFromLegacySecret(s *corev1.Secret) (*NetSpecAuthContext,
 		return &NetSpecAuthContext{}, nil
 	}
 
-	legacyAuth := legacycommonconfig.GetAuthConfigFromSecret(s)
+	legacyAuth := getAuthConfigFromSecret(s)
 	protocolStr, protocolContract := extractProtocol(legacyAuth)
```

Comment:
Maybe not related to your changes but something to track into an issue since here the real goal of `getAuthConfigFromSecret` + `extractProtocol` is to just figure out the protocol, I think we can simplify `getAuthConfigFromSecret` + `extractProtocol` and remove `KafkaAuthConfig` (and its inner structs) to just have a function that given a "legacy channel secret" returns `protocolStr, protocolContract`

Reply:
:+1: I created #2905 as a follow up on this (given that this PR goes through).